main_oneapi.cc

```diff
@@ -67,7 +114,7 @@ int main(int argc, char **argv) {
     oneapi::rawtodigi(input_d, output_d, input.wordCounter, true, true, false, queue);
 
     queue.memcpy((void *)output_h, (void *)output_d, sizeof(Output));
-    queue.wait();
+    queue.wait_and_throw();
```

Comment:
Does the `wait_and_throw()` mean that the in case of errors the `exception_handler()` gets called? (i.e. with plain `wait()` the errors are ignored?)

Reply:
There is a short explanation on the [ComputeCpp migration guide](https://developer.codeplay.com/products/computecpp/ce/guides/sycl-for-cuda-developers/migration):

> In SYCL, error reporting is done via the C++ exception mechanism. Synchronous operations will throw exceptions derived from `cl::sycl::exception` and can be catch in-place by developers, like in the example below:
> ```c++
> try {
>     mySyclProgram.build_with_kernel_type<class myKernel>()
> } catch (cl::sycl::compile_program_error e) {
>     std::cout << " The program failed to build " << std::endl;
> }
> ```
> Asynchronous operations are captured and stored automatically by the SYCL runtime. Multiple exceptions are stored in the order they are captured in a list, of type `cl::sycl::exception_list`. On construction of a queue, an optional functor - the asynchronous handler - can be specified. Whenever a user calls the `wait_and_throw` or `async_throw` methods of the SYCL queue, this functor is called. Developers can use this function to handle custom behavior for dealing with exceptions, such as reducing the multiple exceptions in a list to the most severe one and re-throwing it. **If neither of those methods are called, the exceptions are discarded when the queue object is destroyed.**

Reply:
From an experimental test with oneAPI, it looks like one needs both the `exception_handler` and the call to `wait_and_throw()` - otherwise the exceptions are silently ignored.

Reply:
Ok, thanks.